core/src/main/scala/org/apache/spark/sql/hive/HiveClientUtil.scala

```diff
@@ -121,19 +125,53 @@ class HiveClientUtil(val sparkContext: SparkContext) extends Logging {
    * meta-store that is configured in the hive-site.xml file.
    */
   @transient
-  protected[sql] var client: HiveClient = newClient()
+  protected[sql] var client: HiveClient = newClientWithLogSetting()
 
 
   def closeCurrent(): Unit = {
     Hive.closeCurrent()
   }
 
+  private def newClientWithLogSetting(): HiveClient = {
+    val currentLevel = ClientSharedUtils.converToJavaLogLevel(LogManager.getRootLogger.getLevel)
+    try {
+      ifSmartConn(() => {
```

Comment:
Nothing significant, needed to use that piece of code at several places. Could have done other way also.